modules/gdprEnforcement.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * This module gives publishers extra set of features to enforce individual purposes of TCF v2
+ */
+
+import * as utils from '../src/utils.js';
+import { config } from '../src/config.js';
+import { hasDeviceAccess } from '../src/utils.js';
+import adapterManager, { gdprDataHandler } from '../src/adapterManager.js';
+import find from 'core-js/library/fn/array/find.js';
+import includes from 'core-js/library/fn/array/includes.js';
+import { registerSyncInner } from '../src/adapters/bidderFactory.js';
+import { getHook } from '../src/hook.js';
+import { validateStorageEnforcement } from '../src/storageManager.js';
+
+const purpose1 = 'storage';
+
+let addedDeviceAccessHook = false;
+let enforcementRules;
+
+function getGvlid() {
+  let gvlid;
+  const bidderCode = config.getCurrentBidder();
+  if (bidderCode) {
+    const bidder = adapterManager.getBidAdapter(bidderCode);
+    gvlid = bidder.getSpec().gvlid;
```

Comment:
What will happen here if gvlid is unspecified? Should it be undefined or 0?

Reply:
looks like it would be `undefined` just like if there was no `bidderCode`

Reply:
If no gvlid and gdpr rules are enforced then bidder or user id module will not have access.